File: 1]. DSA/1]. Data Structures/01]. Array/JavaScript/_002)_Insert_Element.js
function insert(arr, position, element) {
    // shift all elements one
    // place to the back until index
    for (i = arr.length; i > position; i--) {
        arr[i] = arr[i - 1];
    }
    // insert the element at the index
    arr[position] = element;
}

let arr = [1, 2, 3, 5];
console.log(`before insertion -> ${arr}`);
insert(arr, 1, -4);
console.log(`After insertion -> ${arr}`);

File: 1]. DSA/1]. Data Structures/01]. Array/Python/_002)_Insert_Element.py
"""
Function to insert a new element at a given position

Takes the position of an array and an element as arguments
"""


def insert(arr, position, element):
    insert = element
    
    for index in range(position):
        moved, arr[index] = arr[index], insert
        insert = moved
    

arr = [1, 2, 3, 5]
print("Array before insertion:", arr)

insert(arr, 1, -4)
print("After insertion:", arr)
